### PR TITLE
added broadcast api call and test

### PR DESCRIPTION
**Pull request by y0sher**

### CODE REVIEW

File: api/grpc_server.go [hunk 1]
```diff
@@ -65,6 +65,16 @@ func (s SpaceMeshGrpcService) SubmitTransaction(ctx context.Context, in *pb.Sign
 	return &pb.SimpleMessage{Value:"ok"}, nil
 }
 
+// P2P API
+
+func (s SpaceMeshGrpcService) Broadcast(ctx context.Context, in *pb.BroadcastMessage) (*pb.SimpleMessage, error) {
```
Comment (zalmen) [inline api/grpc_server.go]: I know it's not part of this PR - but it should be Spacemesh and not SpaceMesh

File: api/pb/api.proto [hunk 2]
```diff
@@ -22,6 +22,11 @@ message SignedTransaction {
     bytes txData = 1;
 }
 
+message BroadcastMessage {
+    string Protocol = 1;
+    bytes Data = 2;
+}
+
 service SpaceMeshService {
```
Comment (zalmen) [inline api/pb/api.proto]: Same comment - should be Spacemesh

**Review by zalmen — COMMENTED** [review]
Currently only messages that are valid according to the specified protocol will be propagated, which isn't what this call is meant to do. You should add a dedicated protocol that will process and approve any message and propagate it

**Comment by y0sher** [comment]
addressed your comments @zalmen

File: app/main.go [hunk 3]
```diff
@@ -242,6 +242,10 @@ func (app *SpacemeshApp) startSpacemesh(cmd *cobra.Command, args []string) {
 	// start p2p services
 	log.Info("Initializing P2P services")
 	swarm, err := p2p.New(Ctx, app.Config.P2P)
+
+	// Gossip messages are validated before propagating. broadcast test protocol isn't
+	api.ApproveAPIGossipMessages(Ctx, swarm) // todo: maybe only enable in tests?
```
Comment (zalmen) [inline app/main.go]: I agree with the TODO, we wouldn't want to allow anyone to spam the network. It should be configurable with a default to false

**Review by zalmen — APPROVED** [review]

**Pull request merged**